plugins/polio/js/src/components/Inputs/DebouncedTextInput.tsx

```diff
@@ -0,0 +1,103 @@
+import { get } from 'lodash';
+import React, {
+    FunctionComponent,
+    useCallback,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { useDebounce } from 'use-debounce';
+// @ts-ignore
+import { useSkipEffectOnMount } from 'bluesquare-components';
+import InputComponent from '../../../../../../hat/assets/js/apps/Iaso/components/forms/InputComponent';
+
+type Props = {
+    field: any;
+    form: any;
+    label: string;
+    required: boolean;
+    debounceTime?: number;
+    withMarginTop?: boolean;
+    clearable?: boolean;
+    // touchOnFocus?: boolean;
+};
+
+export const DebouncedTextInput: FunctionComponent<Props> = ({
+    field,
+    form,
+    label,
+    required,
+    debounceTime = 0,
+    withMarginTop = false,
+    clearable = false,
+    // touchOnFocus = true,
+}) => {
+    const { name } = field;
+    const {
+        setFieldValue,
+        touched,
+        errors: formErrors,
+        setFieldTouched,
+    } = form;
+    const hasError =
+        form.errors &&
+        Boolean(get(form.errors, field.name) && get(touched, field.name));
+    const errors = useMemo(
+        () => (hasError ? [get(formErrors, name)] : []),
+        [formErrors, hasError, name],
+    );
+    const onChange = useCallback(
+        value => {
+            setFieldTouched(name, true);
+            setFieldValue(name, value);
+        },
+        [name, setFieldTouched, setFieldValue],
+    );
+    const parsedValue =
+        typeof field?.value === 'number'
+            ? `${field.value}`
+            : field?.value ?? '';
+    const prevValue = useRef<string>();
+    const prevDebounced = useRef();
+    const [textValue, setTextValue] = useState(parsedValue);
+    const [debouncedValue] = useDebounce(textValue, debounceTime);
+
+    useEffect(() => {
+        prevDebounced.current = parsedValue;
+        // leaving the deps empty because the effect should only run once
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
`useEffect(() => {
        prevDebounced.current = parsedValue;
    });` will do the trick then !